src/database/mod.rs

```diff
@@ -331,6 +333,19 @@ impl Database {
             .collect::<Vec<_>>())
     }
 
+    pub async fn get_latest_root_by_status(&self, status: Status) -> Result<Option<Hash>, Error> {
+        let query = sqlx::query(
+            r#"
+              SELECT root from identities where status = $1 ORDER BY leaf_index DESC LIMIT 1  
```

Comment:
```suggestion
            SELECT root 
            FROM identities
            WHERE status = $1
            ORDER BY id DESC
            LIMIT 1
```

Fixed capitalization and changed `leaf_index` to `id` to the order by. We've introduced id as part of deletion/recovery as now we can have updates essentially at any leaf in the tree.

Can you double check that this works?